src/coreclr/System.Private.CoreLib/src/System/Collections/Generic/EqualityComparer.CoreCLR.cs

```diff
@@ -183,9 +183,34 @@ internal override int LastIndexOf(T[] array, T value, int startIndex, int count)
             int endIndex = startIndex - count + 1;
             for (int i = startIndex; i >= endIndex; i--)
             {
-                if (System.Runtime.CompilerServices.RuntimeHelpers.EnumEquals(array[i], value)) return i;
+                if (RuntimeHelpers.EnumEquals(array[i], value)) return i;
             }
             return -1;
         }
     }
+
+    // Instantiated internally by VM
+    internal sealed class NullableEnumEqualityComparer<T> : EqualityComparer<T?> where T : struct, Enum
```

Comment:
This needs tests and extra code to ensure that the binary serialization continues to work.